packages/@lwc/template-compiler/src/__tests__/fixtures/directive-lwc-if-else/slots/valid-duplicate-slot-names/simple/actual.html

```diff
@@ -0,0 +1,8 @@
+<template>
+    <template lwc:if={condition}>
+        <slot name="conditional-slot"></slot>
+    </template>
+    <template lwc:else>
+        <slot name="conditional-slot"></slot>
+    </template>
+</template>
```

Comment:
This comment brings up an interesting point:

Slots currently don't support directives, including `if:true` and `if:false`, so we failed to handle this use case properly. I've fixed that so we should now throw the appropriate `no directives on slots` error.

That being said, I see no reason why we can't support these new directives on slot elements now, since we're guaranteed to not have any duplicates.

I lean towards relaxing this restriction, but in a separate PR.